drivers/retained_mem/retained_mem_rv3028/retained_mem_rv3028.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2024 ANITRA system s.r.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microcrystal_rv3028_retmem
+
+#include <zephyr/drivers/retained_mem.h>
+#include "rtc_rv3028.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(rv3028, CONFIG_RETAINED_MEM_LOG_LEVEL);
+
+#define RV3028_USER_RAM_SIZE (2 * sizeof(uint8_t))
+
+struct rv3028_retained_mem_config {
+	const struct device *parent;
+};
+
+static ssize_t rv3028_retained_mem_size(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return RV3028_USER_RAM_SIZE;
+}
+
+static int rv3028_retained_mem_read(const struct device *dev, off_t offset, uint8_t *buffer,
+				    size_t size)
+{
+	const struct rv3028_retained_mem_config *config = dev->config;
+	const struct device *rtc_dev = config->parent;
+
+	return rv3028_read_regs(rtc_dev, (RV3028_REG_USER_RAM1 + offset), buffer, size);
+}
+
+static int rv3028_retained_mem_write(const struct device *dev, off_t offset, const uint8_t *buffer,
+				     size_t size)
+{
+	const struct rv3028_retained_mem_config *config = dev->config;
+	const struct device *rtc_dev = config->parent;
+
+	return rv3028_write_regs(rtc_dev, (RV3028_REG_USER_RAM1 + offset), buffer, size);
```

Comment:
These functions would then use the reg, someone might not want an application to look or update the first byte and only use the second, etc.